src/Material/Tooltip.elm

```diff
@@ -0,0 +1,523 @@
+        ([ "target" ] ++ parents ++ [ "nextSibling" ])
+
+    paths =
+      List.map createPath [0..4]
+
+    -- Tries to check if the element is actually a tooltip
+    valid path =
+      isTooltipClass path
+        `Json.andThen`
+          (\res ->
+            if res then
+              at path d
+            else
+              Json.fail ""
+          )
+  in
+    Json.oneOf (List.map valid paths)
+
+
+{-| Checks if the target at path is an actual tooltip
+-}
+isTooltipClass : List (String) -> Json.Decoder Bool
+isTooltipClass path =
+  (at path DOM.className)
+    `Json.andThen`
+      (\class ->
+        if String.contains "mdl-tooltip" class then
+          Json.succeed True
+        else
+          Json.succeed False
+      )
+
+
+{-| Decodes a DOMState from a DOM event
+-}
+stateDecoder : Json.Decoder DOMState
+stateDecoder =
+  Json.object3 DOMState
+    (DOM.target DOM.boundingClientRect)
+    (sibling DOM.offsetWidth)
+    (sibling DOM.offsetHeight)
+
+
+
+-- PROPERTIES
+
+
+{-| Tooltip size
+-}
+type Size
+  = Default
+  | Large
+
+
+{-| Tooltip position relative to the element
+-}
+type Position
+  = Left
+  | Right
+  | Top
+  | Bottom
+
+
+{-| Tooltip config
+-}
+type alias Config =
+  { size : Size
+  , position : Position
+  }
+
+
+{-| Default configuration for tooltip
+-}
+defaultConfig : Config
+defaultConfig =
+  { size = Default
+  , position = Bottom
+  }
+
+
+{-| Properties for Tooltip options.
+-}
+type alias Property m =
+  Options.Property Config m
+
+
+{-| Position the tooltip on the left of the target element
+-}
+left : Property m
+left =
+  Options.set (\options -> { options | position = Left })
+
+
+{-| Position the tooltip on the right of the target element
+-}
+right : Property m
+right =
+  Options.set (\options -> { options | position = Right })
+
+
+{-| Position the tooltip above the target element
+-}
+top : Property m
+top =
+  Options.set (\options -> { options | position = Top })
+
+
+{-| Position the tooltip below the target element
+-}
+bottom : Property m
+bottom =
+  Options.set (\options -> { options | position = Bottom })
+
+
+{-| Tooltip with the default size
+-}
+default : Property m
+default =
+  Options.set (\options -> { options | size = Default })
+
+
+{-| Large tooltip
+-}
+large : Property m
+large =
+  Options.set (\options -> { options | size = Large })
+
+
+{-| Maps a Html.Attribute to Material Property
+-}
+mdl : Attribute a -> Internal.Property b a
+mdl =
+  Internal.Attribute
+
+
+
+-- VIEW
+
+
+{-| Component view.
+-}
+view : (Msg -> m) -> Model -> List (Property m) -> List (Html a) -> Html a
+view lift model options content =
+  let
+    summary =
+      Options.collect defaultConfig options
+
+    config =
+      summary.config
+
+    px : Float -> String
+    px f =
+      (toString f) ++ "px"
+
+    pos =
+      if model.isActive then
+        calculatePos config.position model.domState
+      else
+        defaultPos
+  in
+    Options.styled div
+      (cs "mdl-tooltip"

```

Comment:
No need to manually construct the list with `::`; I think the following is clearer:

    [ cs "mdl-tooltip" 
    , cs "is-active" `when` model.isActive
    , ...